app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/annotation/FillChangeActivity.java

```diff
@@ -0,0 +1,230 @@
+
+  private FillManager fillManager;
+  private Fill fill;
+
+  private MapView mapView;
+
+  private boolean fullAlpha = true;
+  private boolean visible = true;
+  private boolean color = true;
+  private boolean allPoints = true;
+  private boolean holes = false;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // configure initial map state
+    MapboxMapOptions options = new MapboxMapOptions()
+      .attributionTintColor(RED_COLOR)
+      .compassFadesWhenFacingNorth(false)
+      .styleUrl(Style.MAPBOX_STREETS)
+      .camera(new CameraPosition.Builder()
+        .target(new LatLng(45.520486, -122.673541))
+        .zoom(12)
+        .tilt(40)
+        .build());
+
+    // create map
+    mapView = new MapView(this, options);
+    mapView.setId(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+
+    setContentView(mapView);
+  }
+
+  @Override
+  public void onMapReady(@NonNull MapboxMap map) {
+    fillManager = new FillManager(mapView, map, "housenum-label");
+
+    map.setOnPolygonClickListener(polygon -> Toast.makeText(
+      FillChangeActivity.this,
+      "You clicked on polygon with id = " + polygon.getId(),
+      Toast.LENGTH_SHORT
+    ).show());
+
+    fill = fillManager.create(new FillOptions()
+      .withLatLngs(
+        new ArrayList<List<LatLng>>() {{
+          add(STAR_SHAPE_POINTS);
+        }})
+      .withFillColor(ColorUtils.colorToRgbaString(BLUE_COLOR))
+    );
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  protected void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    fillManager.onDestroy();
+    mapView.onDestroy();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    switch (item.getItemId()) {
+      case R.id.action_id_alpha:
+        fullAlpha = !fullAlpha;
+        fill.setFillOpacity(fullAlpha ? FULL_ALPHA : PARTIAL_ALPHA);
+        break;
+      case R.id.action_id_visible:
+        visible = !visible;
+        fill.setFillOpacity(visible ? (fullAlpha ? FULL_ALPHA : PARTIAL_ALPHA) : NO_ALPHA);
+        break;
+      case R.id.action_id_points:
+        allPoints = !allPoints;
+        fill.setLatLngs(allPoints ?
+          new ArrayList<List<LatLng>>() {{
+            add(STAR_SHAPE_POINTS);
+          }} :
+          new ArrayList<List<LatLng>>() {{
```

Comment:
It'd be great to clean up the unnecessary subclasses of the ArrayList in this example.